### PR TITLE
SDIT-1298 Use DSL for discharge to hospital test

### PR DESCRIPTION
**Pull request by andymarke**

### CODE REVIEW

File: src/main/java/uk/gov/justice/hmpps/prison/repository/OffenderDeletionRepository.java [hunk 1]
```diff
@@ -121,18 +121,20 @@ private void deleteBaseRecordOffenderBookingData(final Set<Long> offenderIds, fi
...
-        executeNamedSqlWithBookingIds(OffenderDeletionRepositorySql.OD_DELETE_OFFENDER_IDENTIFYING_MARKS, bookIds);
-        executeNamedSqlWithBookingIds(OffenderDeletionRepositorySql.OD_DELETE_OFFENDER_EXTERNAL_MOVEMENTS, bookIds);
-        executeNamedSqlWithBookingIds(OffenderDeletionRepositorySql.OD_DELETE_OFFENDER_BOOKING_DETAILS, bookIds);
-        executeNamedSqlWithBookingIds(OffenderDeletionRepositorySql.OD_DELETE_OFFENDER_ALERTS, bookIds);
-        deleteOffenderGangAffiliations(bookIds);
-        deleteOffenderNonAssociations(bookIds);
-        deleteOffenderSentCalculations(bookIds);
-        executeNamedSqlWithBookingIds(OffenderDeletionRepositorySql.OD_DELETE_OFFENDER_ASSESSMENT_ITEMS, bookIds);
-        executeNamedSqlWithBookingIds(OffenderDeletionRepositorySql.OD_DELETE_OFFENDER_ASSESSMENTS, bookIds);
+        if (!bookIds.isEmpty()) {
```
Comment (andymarke) [inline src/main/java/uk/gov/justice/hmpps/prison/repository/OffenderDeletionRepository.java:124]: this fell over when there are no bookings

File: src/test/java/uk/gov/justice/hmpps/prison/api/resource/impl/OffenderResourceIntTest_timeline.kt [hunk 2]
```diff
@@ -93,9 +93,7 @@ class OffenderResourceTimelineIntTest : ResourceTest() {
 
       @AfterAll
       fun deletePrisoner() {
-        // for now we can't delete a prisoner with no bookings
-        // we are unlikely to have an issues of leaving this dangling person since
-        // they are not associated with in prison
+        builder.deletePrisoner(prisoner.offenderNo)
```
Comment (andymarke) [inline src/test/java/uk/gov/justice/hmpps/prison/api/resource/impl/OffenderResourceIntTest_timeline.kt:96]: fixed this issue pop deleting offenders with no bookings

File: src/test/java/uk/gov/justice/hmpps/prison/api/resource/impl/OffenderResourceIntTest_dischargeToHospital.kt [hunk 3]
```diff
@@ -705,34 +692,38 @@ class OffenderResourceIntTest_dischargeToHospital : ResourceTest() {
...
+    }
   }
 
   private fun createOffenderWithNoBooking() {
-    OffenderBuilder(bookingBuilders = arrayOf()).save(testDataContext).also {
-      offenderNo = it.offenderNo
-      bookingId = it.bookingId
+    builder.build {
+      offenderNo = offender {
+      }.offenderNo
```
Comment (mikehalmamoj) [inline src/test/java/uk/gov/justice/hmpps/prison/api/resource/impl/OffenderResourceIntTest_dischargeToHospital.kt:705]: I'm not sure about this syntax - it means if there are any children defaulted in the Offender DSL you are overriding them. Might be better to do `offenderNo = offender().offenderNo`?

**Review by mikehalmamoj — APPROVED** [review]

File: src/test/java/uk/gov/justice/hmpps/prison/api/resource/impl/OffenderResourceIntTest_dischargeToHospital.kt (hunk 3 above)
Comment (andymarke) [reply src/test/java/uk/gov/justice/hmpps/prison/api/resource/impl/OffenderResourceIntTest_dischargeToHospital.kt:705]: Not sure I know what you mean? `offender` returns an object which has a field `offenderNo` in

**Pull request merged**